tests/src/JIT/Performance/CodeQuality/Span/SpanBench.cs

```diff
@@ -44,315 +46,392 @@ private sealed class TestClass<T>
+        static bool DoWarmUp; // Main sets this when calling a new benchmark routine
 
-        [MethodImpl(MethodImplOptions.NoInlining)]
-        static void TestFillAllReverseSpan(Span<byte> span)
+
+        // Invoke routine to abstract away the difference between running under xunit-perf vs running from the
+        // command line.  Inner loop to be measured is taken as an Action<int>, and invoked passing the number
+        // of iterations that the inner loop should execute.
+        static void Invoke(Action<int> innerLoop, string nameFormat, params object[] nameArgs)
         {
-            for (int i = span.Length; --i >= 0;) {
-                span[i] = unchecked((byte)i);
+            if (IsXunitInvocation)
+            {
+                foreach (var iteration in Benchmark.Iterations)
+                    using (iteration.StartMeasurement())
+                        innerLoop((int)Benchmark.InnerIterationCount);
             }
-        }
+            else
+            {
+                if (DoWarmUp)
+                {
+                    // Run some warm-up iterations before measuring
+                    innerLoop(InnerIterationCount);
+                    // Clear the flag since we're now warmed up (caller will
+                    // reset it before calling new code)
+                    DoWarmUp = false;
+                }
 
-        [MethodImpl(MethodImplOptions.NoInlining)]
-        static void TestFillAllReverseArray(byte[] data)
-        {
-            for (int i = data.Length; --i >= 0;) {
-                data[i] = unchecked((byte)i);
+                // Now do the timed run of the inner loop.
+                Stopwatch sw = Stopwatch.StartNew();
+                innerLoop(InnerIterationCount);
+                sw.Stop();
+
+                // Print result.
+                string name = String.Format(nameFormat, nameArgs);
+                double timeInMs = sw.Elapsed.TotalMilliseconds;
+                Console.WriteLine("{0}: {1}ms", name, timeInMs);
             }
         }
 
-        static int[] GetUnsortedData()
+        // Helper for the sort tests to get some pseudo-random input
+        static int[] GetUnsortedData(int length)
         {
-            int[] unsortedData = new int[Size];
+            int[] unsortedData = new int[length];
             Random r = new Random(42);
             for (int i = 0; i < unsortedData.Length; ++i)
             {
                 unsortedData[i] = r.Next();
             }
             return unsortedData;
         }
+        #endregion // helpers
 
-        [MethodImpl(MethodImplOptions.NoInlining)]
-        static void TestBubbleSortSpan(Span<int> span)
+        // Tests that implement some vary basic algorithms (fill/sort) over spans and arrays
+        #region Algorithm tests
+
+        #region TestFillAllSpan
+        [Benchmark(InnerIterationCount = FillAllIterations)]
+        [InlineData(DefaultLength)]
+        public static void FillAllSpan(int length)
         {
```

Comment:
I think we can remove this test now since we have a Span.Fill method that does this, no? Or is this meant to test writing into a span as well using the indexer?